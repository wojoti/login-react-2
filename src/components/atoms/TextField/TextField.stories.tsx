/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import TextField from "./TextField";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/TextField",
  component: TextField,
} as ComponentMeta<typeof TextField>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TextField> = (args) => (
  <TextField {...args} />
);

export const Email = Template.bind({});
Email.args = { type: "email", name: "email", id: "email" };

export const Password = Template.bind({});
Password.args = { type: "password", name: "password", id: "password" };
