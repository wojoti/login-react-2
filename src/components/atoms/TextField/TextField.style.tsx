import styled from 'styled-components'

export const StyledTextField = styled.input`
  width: 100%;
  height: 32px;
  border: 2px solid #e5e7eb;
  border-radius: 5px;
  padding-left: 10px;
`
