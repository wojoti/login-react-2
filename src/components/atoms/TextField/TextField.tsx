import { forwardRef } from "react";
import StyledTextField from "./TextField.style";

export interface TextFieldProps {
  type: string;
  name: string;
  id: string;
  testId?: string;
}
export type TextFieldRef = HTMLInputElement;

const TextField = forwardRef<TextFieldRef, TextFieldProps>(
  ({ type, name, id, testId }, ref) => (
    <StyledTextField
      type={type}
      name={name}
      id={id}
      data-testid={testId}
      ref={ref}
    />
  )
);
TextField.displayName = "TextField";

export default TextField;
