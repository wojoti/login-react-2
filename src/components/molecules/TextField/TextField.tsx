import React from 'react'
import Label from '../../atoms/Label/Label'
import TextInput from '../../atoms/TextInput/TextInput'

type Props = {
  type: string
  name: string
  id: string
  color?: string
  onFieldChange?: React.ChangeEventHandler<HTMLInputElement>
}

const TextField = (props: Props) => {
  const ref = React.createRef<HTMLInputElement>()
  return (
    <>
      <Label>{props.name}</Label>
      <TextInput
        type={props.type}
        name={props.name}
        id={props.id}
        onFieldChange={props.onFieldChange}
        ref={ref}
      ></TextInput>
    </>
  )
}

export default TextField
