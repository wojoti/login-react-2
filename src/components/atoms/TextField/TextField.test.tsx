import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";
import TextField, { TextFieldProps } from "./TextField";

afterEach(() => {
  cleanup();
});

const props: TextFieldProps = {
  type: "email",
  name: "email",
  id: "email",
  testId: "test-textfield-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<TextField {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render textfield", () => {
  render(<TextField {...props} />);
  const textfieldElement = screen.getByTestId("test-textfield-id");
  expect(textfieldElement).toBeInTheDocument();
});

test("should handle text change", () => {
  render(<TextField {...props} />);
  const textfieldElement = screen.getByTestId("test-textfield-id");
  expect(textfieldElement).toBeInTheDocument();
  userEvent.type(textfieldElement, "test1234");
  expect(textfieldElement).toHaveProperty("value", "test1234");
});
