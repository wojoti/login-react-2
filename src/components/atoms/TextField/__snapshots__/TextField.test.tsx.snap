// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<input
  className="sc-bcXHqe eKEDvH"
  data-testid="test-textfield-id"
  id="email"
  name="email"
  type="email"
/>
`;
